c7n/resources/cloudtrail.py

```diff
@@ -31,3 +40,93 @@ class resource_type(object):
         name = 'Name'
         dimension = None
         config_type = "AWS::CloudTrail::Trail"
+
+
+policy_json_tmpl = u'''{{
+  "Version": "2012-10-17",
+  "Statement": [
+    {{
+      "Sid": "AWSCloudTrailAclCheck20150319",
+      "Effect": "Allow",
+      "Principal": {{
+        "Service": "cloudtrail.amazonaws.com"
+      }},
+      "Action": "s3:GetBucketAcl",
+      "Resource": "arn:aws:s3:::{bucket_name}"
+    }},
+    {{
+      "Sid": "AWSCloudTrailWrite20150319",
+      "Effect": "Allow",
+      "Principal": {{
+        "Service": "cloudtrail.amazonaws.com"
+      }},
+      "Action": "s3:PutObject",
+      "Resource": "arn:aws:s3:::{bucket_name}/AWSLogs/{account_id}/*",
+      "Condition": {{
+        "StringEquals": {{
+          "s3:x-amz-acl": "bucket-owner-full-control"
+        }}
+      }}
+    }}
+  ]
+}}'''
+
+
+@actions.register('enable')
+class EnableTrail(BaseAction):
+    """Enables logging on the trail(s) named in the policy
+
+    :Example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: trail-test
+            description: Ensure CloudTrail logging is enabled
+            resource: cloudtrail
+            actions:
+              - type: enable
+                trails:
+                - mytrail
+    """
+
+    schema = type_schema(
+        'enable',
+        trails={'type': 'array', 'items': {'type': 'string'}},
+        required=('trails',),
+    )
+
+    def process(self, trails):
+        """Create or enable CloudTrail"""
+        session = local_session(self.manager.session_factory)
+        client = session.client('cloudtrail')
+        existing = {t.get('Name') for t in trails}
+        requested = set(self.data.get('trails'))
+        to_create = requested.difference(existing)
+        to_enable = requested.intersection(existing)
+        for trail in trails:
+            if trail.get('Name') in to_enable:
+                # enable known trails if disabled
+                arn = trail['TrailARN']
+                status = client.get_trail_status(Name=arn)
+                if not status['IsLogging']:
+                    client.start_logging(Name=arn)
+        if len(to_create):
+            account_id = get_account_id(session)
+        else:
+            account_id = ''
+        for trail_name in to_create:
+            # create a trail (and its bucket)
+            bucket_name = '{}-trails'.format(trail_name)
+            s3client = session.client('s3')
+            s3client.create_bucket(Bucket=bucket_name)
+            policy_json = policy_json_tmpl.format(
+                bucket_name=bucket_name, account_id=account_id,
+            )
+            s3client.put_bucket_policy(Bucket=bucket_name, Policy=policy_json)
```

Comment:
good point.. so the end goal is to have something like 'ensure' trail, such that no matter what an active trail is running, that we can trigger off events that modify or disable trails or for initial account setup.